users/src/main/java/engineering/everest/lhotse/users/eventhandlers/UsersEventHandler.java

```diff
@@ -78,6 +83,13 @@ void on(UserPromotedToOrganizationAdminEvent event) {
         usersRepository.save(persistableUser);
     }
 
+    @EventHandler
+    void on(UserDeletedAndForgottenEvent event) {
+        LOGGER.info("Deleting user {}", event.getDeletedUserId());
+        usersRepository.deleteById(event.getDeletedUserId());
```

Comment:
This could be problemtatic. For example, if the deleted user is an organization Admin, its ID will still be referenced in an `OrganizationAggregate`, which will throw error when used. A simple solution could be just rejecting deletion if the user is an org admin. Otherwise, if we try to cacade the changes, it will alsmot call for a saga, which seems to be overkill.

The starterkit is simple enough so a simple rejection validation would work, but I wonder what would be the way to solve this kinda of issue in a large and complex system where validation does not always do the job.

Reply:
I can see a valid use case where an organisation is disabled due to the single org administrator deciding to not use a service and wanting to remove themselves.

This would definitely be a saga - forcing deletion of an organisation so it's not left without someone in charge of it.

It's worth having in there as an example. I'll create a card to do this later.